deflect/StreamPrivate.cpp

```diff
@@ -117,7 +117,9 @@ bool StreamPrivate::finishFrame()
 {
     // Open a window for the PixelStream
     const MessageHeader mh( MESSAGE_TYPE_PIXELSTREAM_FINISH_FRAME, 0, name );
-    return socket.send( mh, QByteArray( ));
+    const bool ret = socket.send( mh, QByteArray( ));
```

Comment:
I played with the flush as well. Imo it does not make any difference as we call waitForBytesWritten() which flushes internally. The problem of the missing frame must come from somewhere else. As it is apparently not even the lack of a QApplication, I'm out of ideas (other than replacing QTCPSocket with ZMQ or similar...).